src/remote.rs

```diff
@@ -588,6 +588,7 @@ impl<'cb> Binding for FetchOptions<'cb> {
             prune: crate::call::convert(&self.prune),
             update_fetchhead: crate::call::convert(&self.update_fetchhead),
             download_tags: crate::call::convert(&self.download_tags),
+            depth: 1, // See `GIT_FETCH_OPTIONS_INIT`.
```

Comment:
I think this needs to be 0. `GIT_FETCH_OPTIONS_INIT` defines the defaults for all the fields up to `proxy_opts`. All fields after that default to 0/null (depth, follow_redirects, and custom_headers). A depth of 1 means to do a shallow clone/fetch.

```suggestion
            depth: 0, // GIT_FETCH_DEPTH_FULL
```

Reply:
I forgot what I was thinking… Anyway, thanks!